src/reactions/MessageReactionList.js

```diff
@@ -0,0 +1,105 @@
+/* @flow strict-local */
+import React, { PureComponent } from 'react';
+import { Dimensions, View } from 'react-native';
+import { TabNavigator, TabBarTop } from 'react-navigation';
+import type { NavigationScreenProp } from 'react-navigation';
+
+import ReactionUserList from './ReactionUserList';
+import { connect } from '../react-redux';
+import type { AggregatedReaction, Dispatch, GlobalState, Message, UserOrBot } from '../types';
+import { Screen, RawLabel } from '../common';
+import { getOwnUser } from '../selectors';
+import aggregateReactions from './aggregateReactions';
+import styles from '../styles';
+import { getAllUsersById } from '../users/userSelectors';
+import tabsOptions from '../styles/tabs';
+import EmojiRow from '../emoji/EmojiRow';
+import { objectFromEntries } from '../jsBackport';
+
+const getReactionsTabs = (
+  aggregatedReactions: $ReadOnlyArray<AggregatedReaction>,
+  initialRouteName: string,
+  usersById: Map<number, UserOrBot>,
+) => {
+  const reactionsTabs = objectFromEntries(
+    aggregatedReactions.map(aggregatedReaction => {
+      const { width } = Dimensions.get('window');
+      const emojiSize: number = Math.min((width / aggregatedReactions.length) * 0.32, 24);
+      const countSize: number = Math.min((width / aggregatedReactions.length) * 0.4, 16);
+      return [
+        aggregatedReaction.name,
+        {
+          screen: () => (
+            <ReactionUserList reactedUserIds={aggregatedReaction.users} users={usersById} />
+          ),
+          navigationOptions: {
+            tabBarLabel: () => (
+              <View style={styles.row}>
+                <EmojiRow name={aggregatedReaction.name} showEmojiName={false} size={emojiSize} />
+                <RawLabel style={{ fontSize: countSize }} text={`  ${aggregatedReaction.count}`} />
+              </View>
+            ),
+          },
+        },
+      ];
+    }),
+  );
+
+  return TabNavigator(reactionsTabs, {
+    backBehavior: 'none',
+    initialRouteName,
+    tabBarComponent: TabBarTop,
+    tabBarPosition: 'top',
+    ...tabsOptions({
+      showLabel: true,
+      showIcon: false,
+      style: {
+        borderWidth: 0.15,
+        borderColor: 'gray',
+      },
+    }),
+  });
+};
+type NavigationsProps = {|
+  navigation: NavigationScreenProp<{ params: {| initialRouteName: string, messageId: number |} }>,
```

Comment:
How about calling this `targetReactionName` throughout (like you do in the webview-to-RN event), instead of `initialRouteName`?

The latter is kind of confusing from the perspective of outside this component -- it only makes sense as sort of an implementation detail, of how the component has chosen to name the "routes" in the TabNavigator it creates within itself. But `targetReactionName` describes the meaning pretty well.